### PR TITLE
[RefManager] - DOI / URL Upload

### PR DESCRIPTION
**Pull request by calvinhlee23**

### CODE REVIEW

File: src/citation/views/citation_entry_view.py [hunk 1]
```diff
@@ -141,6 +149,31 @@ def doi_search(self, request):
         result = open_alex.get_data_from_doi(doi_string)
         return Response(result, status=200)
 
+    @action(detail=False, methods=["get"], permission_classes=[IsAuthenticated])
+    def url_search(self, request):
+        try:
+            url_string = request.query_params.get("url", None)
+            scraper_result = cloudscraper.create_scraper().get(url_string)
```
Comment (le-sun) [inline src/citation/views/citation_entry_view.py:156]: Let's put a timeout inside `.get(...)`

**Review by le-sun — CHANGES_REQUESTED** [review]
Small change

File: src/citation/views/citation_entry_view.py [hunk 2]
```diff
@@ -141,6 +149,31 @@ def doi_search(self, request):
...
+            status_code = scraper_result.status_code
+            dois = []
+
+            if status_code >= 200 and status_code < 400:
+                content = BeautifulSoup(scraper_result.content, "lxml")
+                dois = re.findall(
+                    r"10.\d{4,9}\/[-._;()\/:a-zA-Z0-9]+?(?=[\";%<>\?#&])", str(content)
```
Comment (yattias) [inline src/citation/views/citation_entry_view.py:163]: I've seen DOI regex used in a bunch of locations. Can we store it externally somewhere in a constant?

**Review by le-sun — APPROVED** [review]

File: src/citation/views/citation_entry_view.py [hunk 3]
```diff
@@ -141,6 +148,29 @@ def doi_search(self, request):
...
+            if status_code >= 200 and status_code < 400:
+                content = BeautifulSoup(scraper_result.content, "lxml")
+                dois = re.findall(DOI_REGEX, str(content))
+                parsed_url = urlparse(url_string)
+                cleaned_dois = clean_dois(parsed_url, list(map(str.strip, dois)))
+                doi_counter = Counter(cleaned_dois)
+                formmated_dois = [doi for doi, _ in doi_counter.most_common(1)]
+
+                open_alex = OpenAlex()
+                open_alex_result = open_alex.get_data_from_doi(formmated_dois[0])
```
Comment (le-sun) [inline src/citation/views/citation_entry_view.py:168]: It's possible to have an empty list, so we need to put a check in for that.

**Review by le-sun — CHANGES_REQUESTED** [review]

**Pull request merged**